src/display.cpp

```diff
@@ -1293,13 +1293,11 @@ void display::drawing_buffer_add(const drawing_layer layer, const map_location&
 		int(zoom_)
 	};
 
-	// C++20 needed for in-place aggregate initilization
-#ifdef HAVE_CXX20
+#ifdef __cpp_aggregate_paren_init
```

Comment:
This works on MSVC without C++20?

Reply:
No, it still needs C++20, but we can use these since they're standard macros. More precise feature gating.

Reply:
Ah, so even in C++20 it's just an ordinary macro? Then it makes sense that it could be used in older code where the macro simply wouldn't be defined.

Reply:
Yes. They're listed here: https://en.cppreference.com/w/cpp/feature_test#Library_features